plenum/server/consensus/consensus_shared_data.py

```diff
@@ -1,22 +1,32 @@
 from typing import List
 
+from sortedcontainers import SortedListWithKey
+
 from plenum.common.messages.node_messages import Checkpoint, PrePrepare
 from plenum.server.quorums import Quorums
 
 
-class ConsensusDataProvider:
+class ConsensusSharedData:
     """
     This is a 3PC-state shared between Ordering, Checkpoint and ViewChange services.
     TODO: Consider depending on audit ledger
     TODO: Consider adding persistent local storage for 3PC certificates
     TODO: Restore primary name from audit ledger instead of passing through constructor
     """
-    def __init__(self, name: str, validators: List[str], primary_name: str):
+
+    def __init__(self, name: str, validators: List[str], replica_index):
         self._name = name
-        self.set_validators(validators)
+        self.replica_index = replica_index
```

Comment:
I believe `instance_id` or `inst_id` or `instance_index` would be better here.

Reply:
Actually we have `instId` in other places, so probably `inst_id` here would be the best.

Reply:
ok